services/matrix/matrix.service.js

```diff
@@ -31,15 +36,36 @@ const documentation = `
     <ul>
       <li>Select the desired room inside the Riot.im client</li>
       <li>Click on the room settings button (gear icon) located near the top right of the client</li>
-      <li>Scroll to the very bottom of the settings page and look under the <code>Advanced</code> tab</li>
-      <li>You should see the <code>Internal room ID</code> with your rooms ID next to it (ex: <code>!ltIjvaLydYAWZyihee:matrix.org</code>)</li>
-      <li>Replace the IDs <code>:</code> with <code>/</code></li>
-      <li>The final badge URL should look something like this <code>/matrix/!ltIjvaLydYAWZyihee/matrix.org.svg</code></li>
+      <li>Scroll to the very bottom of the settings page and look under the <code>Addresses</code> section</li>
+      <li>You should see one or more <code>room addresses (or aliases)</code>, which can be easily identified with their starting hash (<code>#</code>) character (ex: <code>#twim:matrix.org</code>)</li>
+      <li>If there is no address for this room, add one under <code>Local addresses for this room</code></li>
+      <li>Remove the starting hash character (<code>#</code>)</li>
+      <li>The final badge URL should look something like this <code>/matrix/twim:matrix.org.svg</code></li>
     </ul>
+    </br>
+    Some Matrix homeservers don't hold a server name matching where they live (e.g. if the homeserver <code>example.com</code> that created the room alias <code>#mysuperroom:example.com</code> lives at <code>matrix.example.com</code>).
+    </br>
+    If that is the case of the homeserver that created the room alias used for generating the badge, you will need to add an extra parameter to the URL, by adding a forward slash followed by the server's FQDN (fully qualified domain name) between the room alias and the <code>.svg</code> extension.
+    </br>
+    The final badge URL should then look something like this <code>/matrix/mysuperroom:example.com/matrix.example.com.svg</code>.
```

Comment:
Another option we have with the API is to put the FQDN in the query string, like the way we handle custom NPM registries. Would that be better?

I don't have an opinion.

Reply:
The [gitlab pipeline status badge](https://github.com/badges/shields/blob/2fe61d2c5cc3d51c641aae588252e05ee72e2d82/services/gitlab/gitlab-pipeline-status.service.js#L15-L32) is a good modern example. Are you up for updating that? I'm good to get this merged after that! Really appreciate all your work on this!

Reply:
Done in https://github.com/badges/shields/pull/2527/commits/410015ea951608ec7e357ed6081d06f84cd816e7

Reply:
Haha, didn't see your messages in the meantime. I based my work on the NPM badges (in a much simpler version though).

Reply:
No prob :) I just pushed a commit with validation, and switching to snake case which for better or worse is what we've used for query params in services.

I'll merge this in a second if all looks good!